### PR TITLE
Unify link colors for visited and hover effects.

### LINKED ISSUE #4274: [Devhub Landing] Add global link hover, visited styles
Supports mozilla/addons-frontend#1021

### PR DESCRIPTION
**Pull request by EnTeQuAk**
Fixes #4274 

We might want to cherry-pick this into this weeks release if we're enabling devhub pages.

Hard to screenshot hover / visited styles, especially since the fading is really subtle. I hope it works like this :) They really just fade a tiny but visible bit.

### CODE REVIEW

File: static/css/devhub/new-landing/base.less [hunk 1]
```diff
@@ -65,6 +65,23 @@ img {
   max-width: 100%;
 }
 
+.link(@color, @decoration: none) {
+  color: @color;
+  text-decoration: @decoration;
+
+  &:visited {
```
Comment (muffinresearch) [inline static/css/devhub/new-landing/base.less]: What about `:link` and `:active`?

File: static/css/devhub/new-landing/base.less [hunk 2]
```diff
@@ -65,6 +65,23 @@ img {
...
+    color: fadeOut(@color, 10%);
+  }
+
+  &:hover {
```
Comment (muffinresearch) [inline static/css/devhub/new-landing/base.less]: I think that non-button standard links should at least show an underline on hover and/or focus.

Comment (muffinresearch) [inline static/css/devhub/new-landing/base.less]: See also https://www.w3.org/TR/2008/NOTE-WCAG20-TECHS-20081211/G183 for more rationale.

File: static/css/devhub/new-landing/base.less (hunk 1 above)
Comment (EnTeQuAk) [inline static/css/devhub/new-landing/base.less]: Good point, I'll add them.

File: static/css/devhub/new-landing/base.less (hunk 2 above)
Comment (EnTeQuAk) [inline static/css/devhub/new-landing/base.less]: Nice reference, thanks! I'll add underlines and refactor the mixin a bit more to be separated from our button.

**Comment by muffinresearch** [comment]
I don't think the contrast for the buttons is going to be enough. It's probably less critical on the links if there's an underline added, but the buttons it makes the text too hard to read against the background.

File: static/css/devhub/new-landing/base.less (hunk 1 above)
Comment (muffinresearch) [inline static/css/devhub/new-landing/base.less]: Also `:focus` is missing.

**Comment by EnTeQuAk** [comment]
> Needs a couple of tweaks, see the inline comments for reference.

I tweaked this quite a bit, I think I'm happy with this now. Buttons darken their background-color and links fade-out a bit + add a underline. Added support for `:focus`, `:active` and `:link` respectively. 

@muffinresearch let me know if this looks / works better.

**Comment by muffinresearch** [comment]
> I tweaked this quite a bit, I think I'm happy with this now. Buttons darken their background-color and links fade-out a bit + add a underline. Added support for :focus, :active and :link respectively.
@muffinresearch let me know if this looks / works better.

Looks much better - the "button" focused style outline is funky! One thing there's no underline on focus when navigating with a keyboard.

**Comment by EnTeQuAk** [comment]
> One thing there's no underline on focus when navigating with a keyboard.

I dismissed that with the reasoning that it looked really weird having the outline (which is still exists) and the underline. If you feel like this shouldn't matter I am happy to add this back.

**Comment by muffinresearch** [comment]
> I dismissed that with the reasoning that it looked really weird having the outline (which is still exists) and the underline. If you feel like this shouldn't matter I am happy to add this back.

Yep fair enough I just tried that and I see what you mean. The outline should amount to a non-colour change so I think that's good enough. Thanks!

**Comment by muffinresearch** [comment]
Remember to squash to aid cherry-picking!

**Pull request merged**